src/boot/efi/util.h

```diff
@@ -216,3 +216,7 @@ static inline bool in_hypervisor(void) {
         return false;
 }
 #endif
+
+static inline bool efi_guid_equal(const EFI_GUID *a, const EFI_GUID *b) {
```

Comment:
I think taking these by value is nicer like in userspace (and would be in line with https://github.com/systemd/systemd/pull/25354)

Reply:
Windows APIs are really inconsistent here, with some functions passing GUIDs by reference and others by value. systemd should choose one way and do it everywhere, rather than making the Windows mistake. Seems like @medhefgo has good reasons for doing by value? So let's just do that and apply it uniformly to every function in the tree?